### ISSUE TITLE
Operation not permitted error

### ISSUE THREAD

**Issue by renaldodev** [Pab450/RunJS-Cracker#4]
I'm facing a permission error on Mac, there's an idea to get over it? And running as a super user doesn't help it!

![image](https://github.com/Pab450/RunJS-Cracker/assets/48330827/94062382-7cc5-4605-9b49-ab462a6c9018)

**Comment by renaldodev** [comment]
I did it all of course, without success. I even tried to create the tmp dir manually didn't work either, within the RunJs.app dir it doesn't allow you to create any kind of file even with a superuser.

**Comment by Pab450** [comment]
I'm running MacOS Ventura 13.5.2 and I have no problem creating a directory in an .app ... Can you tell me the version of your mac ? And the version of your RunJS ? normally with sudo you become the master and nothing is denied...

**Comment by renaldodev** [comment]
I'm running RunJS 2.9.0 over MacOS Ventura 13.6.1.

**Comment by Pab450** [comment]
I confess I'm a bit confused and don't know how to intervene, it doesn't seem to be a problem with the script but rather with your permissions. 

A simple method would be to change the rights : `sudo chmod +x /Applications/RunJS.app/Contents/Resources`

**Comment by renaldodev** [comment]
I already tried to change the dir permissions it is still applied nevertheless it's not possible to create any kind of file within it, It is much awkward for me as well, but I should be able to do anything within the directory with superuser. 

![image](https://github.com/Pab450/RunJS-Cracker/assets/48330827/f6db5418-f1fb-4ad0-a00a-6f57ea6c1a59)

**Comment by renaldodev** [comment]
I had to temporarily disable systems integrity protection to get the installer to work.

**Comment by aayushsapkota9** [comment]
I am using ubantu and getting same error.

**Comment by Pab450** [comment]
> I am using ubantu and getting same error.

Can you try using this command : 

`sudo chmod -R 777 /opt/RunJS/resources && npm run crack`

**Comment by hallstrom91** [comment]
Hello guys! I have installed the crack successfully but iIdo not get the activated version. 
I run version 2.9.0 and when I click the "active" button in the program, it shows as its actived and I can "deactivate" but the key field is locked and reads "undefined". How can i fix this?

**Comment by guillermodiazhdez** [comment]
> Hello guys! I have installed the crack successfully but iIdo not get the activated version. I run version 2.9.0 and when I click the "active" button in the program, it shows as its actived and I can "deactivate" but the key field is locked and reads "undefined". How can i fix this?
> 
> Edit: My bad.. Just updated my version of RunJS and it work perfectly on popOS with this command: sudo chmod -R 777 /opt/RunJS/resources && npm run crack!
> 
> Thank you!

you managed to make it work correctly ?

**Comment by hallstrom91** [comment]
> you managed to make it work correctly ?

Yes, I had the wrong version of RunJS installed, so I changed to the newest and tried the crack again and it worked fine. I have had some trouble with installing node modules like bcrypt but you can fix that by copying the missing files into the correct dir of RunJS for node modules, from another folder where you run npm install blabla.

**Comment by Pab450** [comment]
Should I close this issue or not? I don't know if there is anything useful for people with permission problems? Or I could suggest clearer indications in the readme? I need help guys!

**Comment by alcalino-1978** [comment]
I also had the same issue with version 2.11.0, and the problem was that I didn't have permissions in the terminal to make changes to other apps.

In Privacy & Security > App management, you need to enable the Terminal.
![Captura de pantalla 2024-03-14 a las 7 27 51](https://github.com/Pab450/RunJS-Cracker/assets/114095946/e7055420-ad6b-4bdd-95e1-534174f09cd5)


After doing that, I was able to successfully launch sudo npm run crack without any issues and have the app complete.

macOs: Ventura

**Comment by aayushsapkota9** [comment]
> > I am using ubantu and getting same error.
> 
> Can you try using this command :
> 
> `sudo chmod -R 777 /opt/RunJS/resources && npm run crack`

This one worked for me Thanks
![image](https://github.com/Pab450/RunJS-Cracker/assets/120156314/0cd53b44-1b5d-4149-b54a-0ef97e606df9)

**Comment by OxamaHxn** [comment]
> I also had the same issue with version 2.11.0, and the problem was that I didn't have permissions in the terminal to make changes to other apps.
> 
> In Privacy & Security > App management, you need to enable the Terminal. ![Captura de pantalla 2024-03-14 a las 7 27 51](https://private-user-images.githubusercontent.com/114095946/312730270-e7055420-ad6b-4bdd-95e1-534174f09cd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyNzQyNTMsIm5iZiI6MTcxMzI3Mzk1MywicGF0aCI6Ii8xMTQwOTU5NDYvMzEyNzMwMjcwLWU3MDU1NDIwLWFkNmItNGJkZC05NWUxLTUzNDE3NGYwOWNkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNlQxMzI1NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02OGJhODNkZmYyNWZmM2I2NmYwN2JjNjgxMDdiZWY0ODllNDliYTVjNWRmMzM5MWY4Y2Q3M2ZjNDA0NzNkNDAxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6nHCOY0qEUbVhDmZ_ID8HnCHtxr8dqHLld6Ah7A_Ef0)
> 
> After doing that, I was able to successfully launch sudo npm run crack without any issues and have the app complete.
> 
> macOs: Ventura

Thanks 🤩
